src/com/geekhub/hw3/user/exceptions/UserNotFoundException.java

```diff
@@ -0,0 +1,21 @@
+package com.geekhub.hw3.user.exceptions;
+
+public class UserNotFoundException extends RuntimeException {
```

Comment:
Runtime exception is unchecked Exception and compiller won't say anything about catching it. Better extend from AuthException.